src/signature.c

```diff
@@ -180,6 +180,28 @@ GBytes *cms_sign(GBytes *content, const gchar *certfile, const gchar *keyfile, g
 				"Read zero bytes");
 		goto out;
 	}
+
+	/* keyring was given, perform verification to obtain trust chain */
+	if (r_context()->config->keyring_path) {
+		CMS_ContentInfo *vcms = NULL;
+		X509_STORE *store = NULL;
+		STACK_OF(X509) *verified_chain = NULL;
```

Comment:
Should `store` and `verified_chain` be freed?

Reply:
Ack. Used g_autoptr for these now.